### ISSUE TITLE
`RenderingServer.material_set_param` doesn't work with texture based objects

### ISSUE THREAD

**Issue by matheusmdx** [godotengine/godot#99016]
### Tested versions

- Reproducible in master, 4.4 dev 4, 4.3 stable

### System information

Godot v4.4.dev4 - Windows 10.0.19045 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated AMD Radeon RX 580 2048SP (Advanced Micro Devices, Inc.; 31.0.21921.1000) - AMD Ryzen 5 3600 6-Core Processor (12 threads)

### Issue description

As [this question](https://forum.godotengine.org/t/using-renderingserver-material-set-param-to-set-sampler2d-parameter-not-working/85165)  in Godot forum shows, `RenderingServer.material_set_param` don't allow to set sampler uniforms with a texture object, which diverges from `Material.set_shader_parameter` that works normally with textures as parameter. That happens because `RenderingServer.material_set_param` rejects any `Object` as value:

https://github.com/godotengine/godot/blob/e65a23762b36b564eb94672031f37fdadba72333/servers/rendering/renderer_rd/storage_rd/material_storage.cpp#L2223-L2232

### Steps to reproduce

1. Create a shader with an uniform variable of type Sampler2D and assign this shader to a node
2. Load any image (the Godot icon as example)
3. Try to set the shader uniform variable using `RenderingServer.material_set_param` and the loaded image
4. Godot will fail with error:  `Condition "p_value.get_type() == Variant::OBJECT" is true.`

Or play the MRP below.

### Minimal reproduction project (MRP)

[test-uniform-set-with-rendering-server.zip](https://github.com/user-attachments/files/17691279/test-uniform-set-with-rendering-server.zip)

**Comment by ydeltastar** [comment]
Servers don't use objects. You need to pass a `RID` instead.
```gdscript
RenderingServer.material_set_param($ColorRect.get_material(), "sample", texture.get_rid())
```

**Comment by matheusmdx** [comment]
> Servers don't use Objects. You need to pass a RID instead.

True, if change this behavior is not desired an alternative could be use a more clear error message.
